src/pages/playground/Playground.tsx

```diff
@@ -175,13 +197,61 @@ const Playground: React.FC<PlaygroundProps> = props => {
     }
   };
 
+  const uploadLogs = React.useCallback(() => {
+    const assessmentIDs = getLoggedAssessmentIds();
+    const assessmentLogs = getAssessmentLogs();
+    const playgroundLogs = getPlaygroundLogs();
+
+    if (assessmentLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(
+        assessmentIDs.assessmentId,
+        assessmentIDs.questionId,
+        assessmentLogs
+      );
+    }
+
+    if (playgroundLogs.inputs.length !== 0) {
+      props.handleKeystrokeUpload(playgroundQuestionId, playgroundQuestionId, playgroundLogs);
+    }
+
+    resetPlaygroundLogging(props.sourceChapter, props.externalLibraryName, props.editorValue);
+  }, [props]);
+
+  const uploadPerHour = React.useCallback(() => {
+    const interval = setInterval(() => {
+      uploadLogs();
+    }, oneHourInMilliSeconds);
+    return () => clearInterval(interval);
+  }, [uploadLogs]);
+
+  // Second useEffect called here just in case.
+  React.useEffect(() => {
+    uploadPerHour();
+  }, [uploadPerHour]);
```

Comment:
Just in case isn't a good enough reason to put a chunk of code in. Please consider what needs to be done for correctness.

Reply:
Especially since your just in case looks like this:

```
 public uploadPerHour = () => {
    const interval = setInterval(() => {
      this.uploadLogs();
    }, oneHourInMilliSeconds);
    return () => clearInterval(interval);
  };
```

Reply:
Yea actually as long as we have the following checks whenever we push a log: 

1) If localStorage hit capacity OR 
2) if # of logs is way too big 

We can just send the payload immediately. Then mayeb a timer is not necessary? What is your opinion on this? ><

Reply:
i think timer is okay, just need to make sure there's 1 timer, and it's something like

interval + random offset within interval, otherwise all the uploads will bunch together and nuke the system.